driver/src/contracts/stablex_contract.rs

```diff
@@ -108,20 +114,45 @@ impl StableXContract for StableXContractImpl {
         Ok(Duration::from_secs(remaining_seconds.as_u64()))
     }
 
+    fn get_last_block_for_batch(&self, batch_id: u32) -> Result<Option<u64>> {
+        const BATCH_DURATION: u64 = 300;
+
+        let web3 = self.instance.raw_instance().web3();
+        let get_block = |block_number: BlockNumber| -> Result<_> {
+            web3.eth()
+                .block(block_number.into())
+                .wait()?
+                .ok_or_else(|| anyhow!("block {:?} is missing", block_number))
+        };
+
+        let mut current_block = get_block(BlockNumber::Latest)?;
+        let mut block_number = None;
+        while (batch_id as u64) < current_block.timestamp.as_u64() / BATCH_DURATION
```

Comment:
Can `current_block.timestamp.as_u64() / BATCH_DURATION` become a utility function since we probably use this logic elsewhere to.